src/md_control_module.f90

```diff
@@ -2503,11 +2511,62 @@ subroutine write_md_checkpoint(th, baro)
         end if
       end if
       call io_close(lun)
+
+      if (flag_variable_temperature) then
+        ! Write instantaneous temperature in md.temperature
+        open(unit=lun,file=md_temperature_file,status='replace')
```

Comment:
Here, before opening unit=lun, you should do 
   call io_assing(lun)

Reply:
Added in 9186b16